kolena/dataset/evaluation.py

```
@@ -148,10 +159,15 @@ def download_results(
 
     :param dataset: The name of the dataset.
     :param model: The name of the model.
-    :return: Tuple of DataFrame of datapoints and list of tuples,
-             each containing an evaluation configuration and the corresponding DataFrame of results.
+    :return: Tuple of DataFrame of datapoints and list of [`EvalConfigResults`][kolena.dataset.EvalConfigResults].
     """
     log.info(f"downloading results for model '{model}' on dataset '{dataset}'")
+    existing_dataset = _load_dataset_metadata(dataset)
```

non-blocking: wonder if we want to add a flag like `raise_not_exists` to this internal method to reduce code dupes.

---

Refactored in https://github.com/kolenaIO/kolena/pull/597/commits/246e01bfd3f28c0e3f1be454c613fbfe3a6ec3e5. Let me know what you think. I initially tried to add a boolean flag to the func arg, but we'd still have code dup given that this function returns `Optional[EntityData]`, meaning all usages would still need to do check if the returned val is None.

There's only one place that actually did not need to raise the not found error, so I instead made a `try...except` block there but keep the not found check in this function itself.